### PR TITLE
CDAP-13611 Application metadata non-versioned note

### PR DESCRIPTION
**Pull request by rohitsinha54**
Issue: https://issues.cask.co/browse/CDAP-13611
Build: https://builds.cask.co/browse/CDAP-DQB416-1

### CODE REVIEW

File: cdap-cli/src/main/java/co/cask/cdap/cli/ArgumentName.java [hunk 1]
```diff
@@ -163,6 +163,9 @@ public String toString() {
     "'<namespace-name>.<artifact-name>.<artifact-version>' or " +
     "'<namespace-name>.<app-name>.<app-version>'.\n" +
     "\n" +
+    "Note:  Metadata for versioned entities such as applications, programs, schedules and program runs is not " +
+    "versioned. If a metadata is added to one specific version it will be reflected to all versions." +
```
Comment (anwar6953) [inline cdap-cli/src/main/java/co/cask/cdap/cli/ArgumentName.java]: "If a metadata" --> "If metadata".
Same in the other file.

File: cdap-cli/src/main/java/co/cask/cdap/cli/ArgumentName.java [hunk 2]
```diff
@@ -163,6 +163,9 @@ public String toString() {
     "'<namespace-name>.<artifact-name>.<artifact-version>' or " +
     "'<namespace-name>.<app-name>.<app-version>'.\n" +
     "\n" +
+    "Note:  Metadata for versioned entities such as applications, programs, schedules and program runs is not " +
```
Comment (deb-saunders) [inline cdap-cli/src/main/java/co/cask/cdap/cli/ArgumentName.java]: Suggest:

**Note**: Metadata for versioned entities is not versioned, including entities such as applications, programs, schedules, and program runs. Additions to metadata in one version are reflected in all versions.

File: cdap-docs/reference-manual/source/http-restful-api/metadata.rst [hunk 3]
```diff
@@ -15,6 +15,9 @@ Metadata HTTP RESTful API
 Use the CDAP Metadata HTTP RESTful API to set, retrieve, and delete the metadata annotations
 of applications, datasets, streams, and other entities in CDAP.
 
+**Note**: Metadata for versioned entities such as applications, programs, schedules and program runs is not versioned.
```
Comment (deb-saunders) [inline cdap-docs/reference-manual/source/http-restful-api/metadata.rst]: Suggest:

**Note**: Metadata for versioned entities is not versioned, including entities such as applications, programs, schedules, and program runs. Additions to metadata in one version are reflected in all versions.

**Review by deb-saunders — CHANGES_REQUESTED** [review]

File: cdap-cli/src/main/java/co/cask/cdap/cli/ArgumentName.java (hunk 2 above)
Comment (rohitsinha54) [reply cdap-cli/src/main/java/co/cask/cdap/cli/ArgumentName.java]: Done.

File: cdap-cli/src/main/java/co/cask/cdap/cli/ArgumentName.java (hunk 1 above)
Comment (rohitsinha54) [reply cdap-cli/src/main/java/co/cask/cdap/cli/ArgumentName.java]: Not needed with above change.

File: cdap-docs/reference-manual/source/http-restful-api/metadata.rst (hunk 3 above)
Comment (rohitsinha54) [reply cdap-docs/reference-manual/source/http-restful-api/metadata.rst]: Done.

**Review by rohitsinha54 — COMMENTED** [review]
@deb-saunders : Thanks for the review. Incorporated all suggestions in second comments. Please have another look.

**Review by anwar6953 — APPROVED** [review]

**Pull request merged**